src/Model/GroundWaterFlow/gwf3vsc8.f90

```diff
@@ -0,0 +1,1519 @@
+                               this%a3, this%a4)
+
+    end do
+    !
+    ! -- Return
+    return
+  end subroutine vsc_calcvisc
+
+  subroutine allocate_scalars(this)
+! ******************************************************************************
+! allocate_scalars
+! ******************************************************************************
+!
+!    SPECIFICATIONS:
+! ------------------------------------------------------------------------------
+    ! -- modules
+    use ConstantsModule, only: DZERO
+    ! -- dummy
+    class(GwfVscType) :: this
+    ! -- local
+! ------------------------------------------------------------------------------
+    !
+    ! -- allocate scalars in NumericalPackageType
+    call this%NumericalPackageType%allocate_scalars()
+    !
+    ! -- Allocate
+    call mem_allocate(this%thermivisc, 'THERMIVISC', this%memoryPath)
+    call mem_allocate(this%idxtmpr, 'IDXTMPR', this%memoryPath)
+    call mem_allocate(this%ioutvisc, 'IOUTVISC', this%memoryPath)
+    call mem_allocate(this%ireadelev, 'IREADELEV', this%memoryPath)
+    call mem_allocate(this%iconcset, 'ICONCSET', this%memoryPath)
+    call mem_allocate(this%viscref, 'VISCREF', this%memoryPath)
+    call mem_allocate(this%a2, 'A2', this%memoryPath)
+    call mem_allocate(this%a3, 'A3', this%memoryPath)
+    call mem_allocate(this%a4, 'A4', this%memoryPath)
+    !
+    call mem_allocate(this%nviscspecies, 'NVISCSPECIES', this%memoryPath)
+    !
+    ! -- Initialize
+    this%thermivisc = 0
+    this%idxtmpr = 0
+    this%ioutvisc = 0
+    this%ireadelev = 0
+    this%iconcset = 0
+    this%viscref = 1000.d0
+    this%A2 = DZERO
+    this%A3 = DZERO
+    this%A4 = DZERO
+    !
+    this%nviscspecies = 0
+    !
+    ! -- Return
+    return
+  end subroutine allocate_scalars
+
+  subroutine allocate_arrays(this, nodes)
+! ******************************************************************************
+! allocate_arrays
+! ******************************************************************************
+!
+!    SPECIFICATIONS:
+! ------------------------------------------------------------------------------
+    ! -- modules
+    ! -- dummy
+    class(GwfVscType) :: this
+    integer(I4B), intent(in) :: nodes
+    ! -- local
+    integer(I4B) :: i
+! ------------------------------------------------------------------------------
+    !
+    ! -- Allocate
+    call mem_allocate(this%visc, nodes, 'VISC', this%memoryPath)
+    call mem_allocate(this%ivisc, this%nviscspecies, 'IVISC', this%memoryPath)
+    call mem_allocate(this%dviscdc, this%nviscspecies, 'DRHODC', &
+                      this%memoryPath)
+    call mem_allocate(this%cviscref, this%nviscspecies, 'CRHOREF', &
+                      this%memoryPath)
+    call mem_allocate(this%ctemp, this%nviscspecies, 'CTEMP', this%memoryPath)
+    allocate (this%cmodelname(this%nviscspecies))
+    allocate (this%cauxspeciesname(this%nviscspecies))
+    allocate (this%modelconc(this%nviscspecies))
```

Comment:
Strings can now be stored in the memory manager.

Reply:
@langevin-usgs, I need a consult with this one.  I attempted to implement a fix based on what I was seeing inside BoundaryPackage.f90.  However, because `gwf3vsc8.f90/read_packagedata()` echos what it reads to the standard listing file, I got to a point where the compiler was complaining:

```
error #5521: A derived-type object in an input/output list cannot have inaccessible components unless a suitable user-defined input/output procedure is available.
error #5514: A derived type I/O list item that contains a pointer or an allocatable component (CHARSTRING) requires a user-defined derived-type input/output procedure.
```
Based on some quick googling, it looks like the fix could be something along [these lines](https://stackoverflow.com/questions/38780389/how-to-write-an-allocatable-array-in-a-derived-type-using-namelists). However, it looks like you anticipated this issue based on [lines 35-37 in CharString.f90](https://github.com/MODFLOW-USGS/modflow6/blob/develop/src/Utilities/CharString.f90), but based on the comment there it would seem to trip a failure in the CI routines?  

There are two options, I think:  

1. comment-out the code that echos the variable to the standard listing file, which I think would mean I could continue implementing the MemoryManager way of doing things for character strings.
2. leave it the way it is for the time being, which essentially mirrors what's in BUY.

I'll proceed with option 2 until advised otherwise

Reply:
Okay, sounds good.  The CHARSTRING implementation might need some additional tending for general use in the program.